File: Tema_2/RepasoRecyclerViewTabLayout/app/src/main/java/com/example/repasorecyclerviewtablayout/ui/home/HomeFragment.java
package com.example.repasorecyclerviewtablayout.ui.home;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProvider;
import androidx.viewpager2.widget.ViewPager2;

import com.example.repasorecyclerviewtablayout.databinding.FragmentHomeBinding;
import com.google.android.material.tabs.TabLayout;
import com.google.android.material.tabs.TabLayoutMediator;

public class HomeFragment extends Fragment {

    private FragmentHomeBinding binding;

    public View onCreateView(@NonNull LayoutInflater inflater,
                             ViewGroup container, Bundle savedInstanceState) {
        HomeViewModel homeViewModel =
                new ViewModelProvider(this).get(HomeViewModel.class);

        binding = FragmentHomeBinding.inflate(inflater, container, false);
        View root = binding.getRoot();

        ViewPager2 viewPager2=binding.pager;
        TabLayout tabLayout=binding.tabLayout;
        ViewPagerAdapter pagerAdapter = new ViewPagerAdapter(this.getActivity());
        viewPager2.setAdapter(pagerAdapter);
        TabLayoutMediator tabLayoutMediator=new TabLayoutMediator(tabLayout,viewPager2,
                (tab,position)->{
                    switch (position) {
                        case 0:
                            tab.setText("Ver Pilotos");
                            break;
                        case 1:
                            tab.setText("Ver Circuitos");
                            break;
                    }
                });
        tabLayoutMediator.attach();
        return root;
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        binding = null;
    }
}

File: Proyecto_Integrado/MiniChef/app/src/main/java/com/example/minichef_v1/pantanllas/home/HomeFragment.kt
package com.example.minichef_v1.pantanllas.home

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.example.minichef_v1.R
import com.example.minichef_v1.databinding.FragmentHomeBinding
import com.example.minichef_v1.pantanllas.home.viewPager2.ViewPager2Adapter
import com.google.android.material.tabs.TabLayoutMediator

class HomeFragment : Fragment() {

    private var _binding: FragmentHomeBinding? = null
    private val binding get() = _binding!!
    private lateinit var adapter: ViewPager2Adapter


    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        _binding = FragmentHomeBinding.inflate(inflater, container, false)
        val root: View = binding.root

        adapter=ViewPager2Adapter(this)

        binding.pager.adapter=adapter
        val tabLayoutMediator=TabLayoutMediator(binding.tabLayout,binding.pager
        ) { tab, position ->
            when (position) {
                0 -> {
                    tab.text = resources.getText(R.string.siguiendo)
                }
                1 -> {
                    tab.text = resources.getText(R.string.populares)
                }
            }
        }
        tabLayoutMediator.attach()

        return root
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }
}